compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.test.cpp

```diff
@@ -0,0 +1,273 @@
+
+  luci::SubstituteStridedSliceToReshapePass pass;
+  while (pass.run(&g))
+    ;
+
+  auto reshape_node = dynamic_cast<luci::CircleReshape *>(output->from());
+  ASSERT_TRUE(reshape_node != nullptr);
+
+  auto new_shape = loco::must_cast<luci::CircleConst *>(reshape_node->shape());
+  ASSERT_EQ(new_shape->rank(), 1);
+  ASSERT_EQ(new_shape->dim(0).value(), 3);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(0), 1);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(1), 5);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(2), 9);
+}
+
+TEST_F(SubstituteStridedSliceToReshapeTest, with_begin_end_mask)
+{
+  buildGraph({5, 1, 9}, // input shape
+             {0, 0, 5}, // begin
+             {3, 1, 9}, // end
+             {1, 1, 1}, // strides
+             0b100,     // begin mask
+             0b001,     // end mask
+             0,         // ellipsis axis mask
+             0,         // new axis mask
+             0b010      // shrink axis mask, 0th and 3rd dim will be shrunk
+  );
+
+  luci::SubstituteStridedSliceToReshapePass pass;
+  while (pass.run(&g))
+    ;
+
+  auto reshape_node = dynamic_cast<luci::CircleReshape *>(output->from());
+  ASSERT_TRUE(reshape_node != nullptr);
+
+  auto new_shape = loco::must_cast<luci::CircleConst *>(reshape_node->shape());
+  ASSERT_EQ(new_shape->rank(), 1);
+  ASSERT_EQ(new_shape->dim(0).value(), 2);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(0), 5);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(1), 9);
+}
+
+TEST_F(SubstituteStridedSliceToReshapeTest, with_large_end_mask)
+{
+  buildGraph({5, 1, 9},       // input shape
+             {0, 0, 0},       // begin
+             {100, 100, 100}, // large end
+             {1, 1, 1},       // strides
+             0,               // begin mask
+             0,               // end mask
+             0,               // ellipsis axis mask
+             0,               // new axis mask
+             0b010            // shrink axis mask, 0th and 3rd dim will be shrunk
+  );
+
+  luci::SubstituteStridedSliceToReshapePass pass;
+  while (pass.run(&g))
+    ;
+
+  auto reshape_node = dynamic_cast<luci::CircleReshape *>(output->from());
+  ASSERT_TRUE(reshape_node != nullptr);
+
+  auto new_shape = loco::must_cast<luci::CircleConst *>(reshape_node->shape());
+  ASSERT_EQ(new_shape->rank(), 1);
+  ASSERT_EQ(new_shape->dim(0).value(), 2);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(0), 5);
+  ASSERT_EQ(new_shape->at<loco::DataType::S32>(1), 9);
+}
+
+TEST_F(SubstituteStridedSliceToReshapeTest, not_matching_begin_index_NEG)
+{
+  buildGraph({1, 3, 5, 7}, // input shape
+             {0, 0, 2, 0}, // begin[2] does not start from 0
+             {1, 3, 5, 7}, // end
+             {1, 1, 1, 1}, // strides
+             0,            // begin mask
+             0,            // end mask
+             0,            // ellipsis axis mask
+             0,            // new axis mask
+             0b0001        // shrink axis mask
+  );
+
+  assert_not_converted();
```

Comment:
I'm not sure but this will make a warning by our analysis tool as there is no gtest defined macro in this function.
- our static analysis tool is not that smart.
- on way to overcome this is to add `SUCCEED();` in each test function.